File: core/models/inventory.py
from datetime import datetime
from decimal import Decimal # For precise quantity representation
from typing import Optional
from pydantic import BaseModel, Field, ConfigDict

from core.models.enums import InventoryMovementType

class InventoryMovement(BaseModel):
    id: Optional[int] = None
    product_id: int
    quantity: Decimal = Field(..., max_digits=10, decimal_places=2) # Positive for in, negative for out
    movement_type: InventoryMovementType = Field(...) # Type of inventory movement
    timestamp: datetime = Field(default_factory=datetime.utcnow) # Renamed from 'date' for clarity and consistency
    description: Optional[str] = Field(default=None, max_length=255)
    user_id: Optional[int] = None # User performing the movement
    related_id: Optional[int] = None # e.g., Sale ID, Purchase ID

    model_config = ConfigDict(from_attributes=True)

File: csharp/src/Eleventa.Domain/Entities/Inventory.cs
namespace Eleventa.Domain.Entities;

/// <summary>
/// Represents an inventory movement record.
/// </summary>
public class InventoryMovement
{
    public int Id { get; set; }
    public int ProductId { get; set; }
    public InventoryMovementType MovementType { get; set; }
    public decimal Quantity { get; set; }
    public decimal? UnitCost { get; set; }
    public string? Reason { get; set; }
    public string? Reference { get; set; }  // Reference to sale, purchase order, etc.
    public int? UserId { get; set; }
    public DateTime Timestamp { get; set; } = DateTime.Now;
    public decimal? PreviousStock { get; set; }
    public decimal? NewStock { get; set; }

    // Navigation properties
    public Product? Product { get; set; }
    public User? User { get; set; }
}
